pkg/api/vminfo.go

```diff
@@ -82,6 +82,14 @@ type VmConfig struct {
 	ScalingConfig  *ScalingConfig `json:"scalingConfig,omitempty"`
 }
 
+// Equals returns true if the two VmConfig structs are deep-equal to each other.
+func (c VmConfig) Equals(other VmConfig) bool {
```

Comment:
Maybe `DeepEqual` [1] or serialize to json and compare jsons? There comparators aren't very pretty + easy to forget when adding new fields.


[1]: https://pkg.go.dev/reflect#DeepEqual

Reply:
I mostly wanted to avoid that if possible, because we fetch and compare the `VmInfo` every time we receive an Update event, and I thought it'd be too inefficient to use something like that.

Maybe `reflect.DeepEqual` is the right call though... I'll pull some stats from production to see what the actual event rate is.

Reply:
Stats from prod: Seems like us-west-2 peaks the highest, at about 3k pod add/update events per minute. I don't know what the second-to-second events load is like, but presumably it's similarly spiky to what we see from neonvm-controller (albeit with some flattening because a spike of VM starts has many subsequent pod events).

All that said, I think we can get away with `reflect.DeepEqual`.

Reply:
Switched to `reflect.DeepEqual`, resolving